src/components/tic-tac-toe.tsx

```diff
@@ -0,0 +1,377 @@
+import * as React from 'react';
+import './ticTacToe.css';
+import { css } from 'emotion';
+
+interface SquareProps {
+  isWinner: boolean;
+  onClick: () => void;
+  value: string;
+}
+
+interface RowProps {
+  rowNum: number;
+  squares: string[];
+  winnerSquares: number[];
+  onClick: (i: number) => void;
+}
+
+interface BoardProps {
+  squares: string[];
+  winnerSquares: number[];
+  onClick: (i: number) => void;
+}
+
+interface ToggleProps {
+  onClick: () => void;
+  isToggleOn: boolean;
+}
+
+interface History {
+  lastMove: number;
+  squares: string[];
+}
+
+interface GameState {
+  history: History[];
+  isToggleOn: boolean;
+  stepNumber: number;
+  xIsNext: boolean;
+}
+
+const Square = (props: SquareProps) => {
+  let image = '';

```

Comment:
This would be much neater using ternary operators + template strings rather than stringbuilding.